openedx_tagging/core/tagging/models.py

```diff
@@ -402,7 +434,16 @@ def is_valid(self) -> bool:
 
         A valid ObjectTag must be linked to a Taxonomy, and be a valid tag in that taxonomy.
         """
-        return self.taxonomy_id and self.taxonomy.validate_object_tag(self)
+        if self.taxonomy_id:
+            # self.taxonomy doesn't return the correct subclass, so we must fetch it
+            # FIXME:
+            # * cache to prevent re-fetching the taxonomy
+            # * discourage people from using self.taxonomy
+            taxonomy = (
+                Taxonomy.objects.filter(id=self.taxonomy_id).select_subclasses().first()
+            )
```

Comment:
What's the best way to do this?  We can:

* cache the fetched taxonomy subclass using a getter property, and
* rename the `taxonomy` field to `_taxonomy`

Reply:
:+1: for this way